src/components/QrcodeCapture.vue

```diff
@@ -13,18 +13,56 @@
       @change="onFileSelect"
     />
 
-    <img ref="imageElement" :src="imageBase64" :class="`${className}__image`" />
+    <canvas ref="canvasElement" :class="`${className}__image`" />
   </div>
 </template>
 
 <script>
 import { vibrate } from '@/lib/vibrate'
+import { useTemplateRef } from 'vue'
 
 export default {
   emits: ['detect', 'error'],
+  setup() {
+    const canvas = useTemplateRef('canvasElement')
+
+    const IMG_MAX_SIZE = 400
```

Comment:
Move outside Vue component

Reply:
I meant above the Vue component definition, not in another file. If you converted to script setup, it's fine.